src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -92,113 +93,116 @@ public FilingsService(
   }
 
   public FilingApi generateOverseasEntityFiling(
-      String requestId,
-      String overseasEntityId,
-      Transaction transaction,
-      String passThroughTokenHeader)
-      throws SubmissionNotFoundException, ServiceException {
+          String requestId,
+          String overseasEntityId,
+          Transaction transaction,
+          String passThroughTokenHeader)
+          throws SubmissionNotFoundException, ServiceException {
     var filing = new FilingApi();
     filing.setKind(FILING_KIND_OVERSEAS_ENTITY);
     setFilingApiData(filing, requestId, overseasEntityId, transaction, passThroughTokenHeader);
     return filing;
   }
 
   private void setFilingApiData(
-      FilingApi filing,
-      String requestId,
-      String overseasEntityId,
-      Transaction transaction,
-      String passThroughTokenHeader)
-      throws SubmissionNotFoundException, ServiceException {
+          FilingApi filing,
+          String requestId,
+          String overseasEntityId,
+          Transaction transaction,
+          String passThroughTokenHeader)
+          throws SubmissionNotFoundException, ServiceException {
     var logMap = new HashMap<String, Object>();
     logMap.put(OVERSEAS_ENTITY_ID_KEY, overseasEntityId);
     logMap.put(TRANSACTION_ID_KEY, transaction.getId());
 
     Map<String, Object> userSubmission = new HashMap<>();
 
+//    OverseasEntitySubmissionDto submissionDto = setSubmissionData(userSubmission, overseasEntityId, logMap);
     OverseasEntitySubmissionDto submissionDto =
-        setSubmissionData(userSubmission, overseasEntityId, logMap);
-    setPaymentData(userSubmission, transaction, passThroughTokenHeader, logMap);
+            overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId)
+                    .orElseThrow(() -> new SubmissionNotFoundException(
+                            String.format("Empty submission returned when generating filing for %s", overseasEntityId)
+                    ));
 
     if (submissionDto.isForUpdate()) {
-      publicDataRetrievalService.initialisePublicData(
-          submissionDto.getEntityNumber(), passThroughTokenHeader);
+      publicDataRetrievalService.initialisePublicData(submissionDto.getEntityNumber(), passThroughTokenHeader);
       privateDataRetrievalService.initialisePrivateData(submissionDto.getEntityNumber());
+      UpdateSubmission updateSubmission = new UpdateSubmission();
+      setUpdateSubmissionData(updateSubmission, submissionDto, logMap);
+      Map<String, Object> updateSubmissionMap = objectMapper.convertValue(updateSubmission, new TypeReference<Map<String, Object>>() {});
+      filing.setData(updateSubmissionMap);
+      filing.setKind(FILING_KIND_OVERSEAS_ENTITY_UPDATE);
+    } else {
+      setSubmissionData(userSubmission, submissionDto, logMap);
+      filing.setData(userSubmission);
```

Comment:
where is filing.setKind() called for registration? I'm a bit concerned that we've changed the flow for registration - please make sure nothing is changed on that path.